x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java

```diff
@@ -263,43 +298,62 @@ public void getCheckpointingInfo(long lastCheckpointNumber, TransformIndexerPosi
+                        "Failed to retrieve source checkpoint for transform [{}]",
+                        transformConfig.getId()
+                    ),
+                    e
+                );
+                listener.onFailure(new CheckpointException("Failure during source checkpoint info retrieval", e));
+            }
+        );
 
         // <2> got the next checkpoint, get the source checkpoint
         ActionListener<TransformCheckpoint> nextCheckpointListener = ActionListener.wrap(
-                nextCheckpointObj -> {
-                    checkpointingInfoBuilder.setNextCheckpoint(nextCheckpointObj);
-                    getIndexCheckpoints(checkpointsByIndexListener);
-                },
-                e -> {
-                    logger.debug((Supplier<?>) () -> new ParameterizedMessage(
-                            "Failed to retrieve next checkpoint [{}] for transform [{}]", lastCheckpointNumber + 1,
-                            transformConfig.getId()), e);
-                    listener.onFailure(new CheckpointException("Failure during next checkpoint info retrieval", e));
-                }
-            );
+            nextCheckpointObj -> {
+                checkpointingInfoBuilder.setNextCheckpoint(nextCheckpointObj);
+                getIndexCheckpoints(checkpointsByIndexListener);
+            },
+            e -> {
+                logger.debug(
+                    (Supplier<?>) () -> new ParameterizedMessage(
+                        "Failed to retrieve next checkpoint [{}] for transform [{}]",
+                        lastCheckpointNumber + 1,
+                        transformConfig.getId()
+                    ),
+                    e
+                );
+                listener.onFailure(new CheckpointException("Failure during next checkpoint info retrieval", e));
+            }
+        );
 
         // <1> got last checkpoint, get the next checkpoint
         ActionListener<TransformCheckpoint> lastCheckpointListener = ActionListener.wrap(
             lastCheckpointObj -> {
                 checkpointingInfoBuilder.lastCheckpoint = lastCheckpointObj;
-                transformConfigManager.getTransformCheckpoint(transformConfig.getId(), lastCheckpointNumber + 1,
-                        nextCheckpointListener);
+                transformConfigManager.getTransformCheckpoint(
+                    transformConfig.getId(),
+                    lastCheckpointNumber + 1,
+                    nextCheckpointListener
+                );
             },
             e -> {
-                logger.debug((Supplier<?>) () -> new ParameterizedMessage(
-                        "Failed to retrieve last checkpoint [{}] for transform [{}]", lastCheckpointNumber,
-                        transformConfig.getId()), e);
+                logger.debug(
+                    (Supplier<?>) () -> new ParameterizedMessage(
+                        "Failed to retrieve last checkpoint [{}] for transform [{}]",
```

Comment:
```suggestion
                        "[{}] failed to retrieve last checkpoint [{}]",
```
And flip the parameters.